controllers/utils.go

```diff
@@ -176,3 +182,49 @@ func ContainerExec[T any](client T, pod *corev1.Pod, command []string, container
 	time.Sleep(time.Second)
 	return result, nil
 }
+
+// customZapLogger should be used only for warn messages
```

Comment:
does it break if we use `.Info` or `.Error`?

Reply:
`Info`, `Debug` and `Error` are allowed by logger:
https://sdk.operatorframework.io/docs/building-operators/golang/references/logging/#default-zap-logger

For `Warn`, `Panic`, `Fatal` or other levels we would need to use `CustomZapLogger`.

Reply:
why not using the `CustomZapLogger` for everything?

Reply:
To avoid breaking compatibility with controller-runtime/logr, which uses a custom zap logger:
https://github.com/pulp/pulp-operator/blob/c5d840d04976be8e67c79818838d770668e537a3/main.go#L39
https://sdk.operatorframework.io/docs/building-operators/golang/references/logging/#non-default-logging

We could try to use zap directly but, from what I could understand, it is not recommended because of the lack of integration with controller-runtime (which interacts with logr)...